cross build regressed wrt System.Native and "shm_open that works well enough with mmap"

<!--
If you are new to the project get yourself familiar with https://www.mono-project.com/community/bugs/make-a-good-bug-report/ before filling the issue
-->

## Steps to Reproduce

1. on e.g. linux amd64: 
`./autogen.sh  CFLAGS="-g -O0" -disable-nls -disable-system-aot --host=arm-linux-gnueabihf CC=arm-linux-gnueabihf-gcc -enable-cross-compiling`

### Current Behavior

```
.
.
.
checking for shm_open that works well enough with mmap... configure: error: in `/s/mono3':
configure: error: cannot run test program while cross compiling
See `config.log' for more details
```

### Expected Behavior

Uncertain. Perhaps success and compile and link, perhaps not.
Maybe cross builds too esoteric.

We have defaults in some places and don't run programs.

```
      dnl ***********************************
        dnl *** Checks for working __thread ***
        dnl ***********************************
        AC_MSG_CHECKING(for working __thread)
        if test "x$with_tls" != "x__thread"; then
                AC_MSG_RESULT(disabled)
        elif test "x$cross_compiling" = "xyes"; then
                AC_MSG_RESULT(cross compiling, assuming yes)
        else
                AC_TRY_RUN([
  
```


and

```
        dnl **************************************
        dnl *** Checks for working sigaltstack ***
        dnl **************************************
        AC_MSG_CHECKING(for working sigaltstack)
        if test "x$with_sigaltstack" != "xyes"; then
                AC_MSG_RESULT(disabled)
        elif test "x$cross_compiling" = "xyes"; then
                AC_MSG_RESULT(cross compiling, assuming yes)
        else
                AC_TRY_RUN([
```



but:

```
       AC_MSG_CHECKING(for shm_open that works well enough with mmap)
        if test "x$ac_cv_func_shm_open" = "xno" -o "x$ac_cv_func_shm_open_working_with_mmap" = "xno" ; then
                AC_MSG_RESULT(no)
        else
                AC_TRY_RUN([
 ```


therefore:

```
checking for shm_open that works well enough with mmap... configure: error: in `/s/mono3':
configure: error: cannot run test program while cross compiling
```

Is there a good default maybe?
This came in prematurely with System.Native ahead of its memory mapped file support?
Which is coming?


Anyway, I'll just set ac_cv_func_shm_open_working_with_mmap and move on.


## On which platforms did you notice this

Linux

--- a/configure.ac
+++ b/configure.ac
@@ -3301,6 +3301,7 @@ if test x$host_win32 = xno; then
 		AC_MSG_RESULT(no)
 	elif test "x$cross_compiling" = "xyes"; then
 		AC_MSG_RESULT(cross compiling, assuming yes)
+		AC_DEFINE(HAVE_SHM_OPEN_THAT_WORKS_WELL_ENOUGH_WITH_MMAP, 1, [shm_open that works well enough with mmap])
 	else
 		AC_TRY_RUN([
 			#include <sys/mman.h>

--- a/configure.ac
+++ b/configure.ac
@@ -3299,6 +3299,8 @@ if test x$host_win32 = xno; then
 	AC_MSG_CHECKING(for shm_open that works well enough with mmap)
 	if test "x$ac_cv_func_shm_open" = "xno" -o "x$ac_cv_func_shm_open_working_with_mmap" = "xno" ; then
 		AC_MSG_RESULT(no)
+	elif test "x$cross_compiling" = "xyes"; then
+		AC_MSG_RESULT(cross compiling, assuming yes)
 	else
 		AC_TRY_RUN([
 			#include <sys/mman.h>

